src/ocean_emulators/models/modules/blocks.py

```diff
@@ -46,6 +46,49 @@ def forward(self, x):
         return self.upsampler(x)
 
 
+class ZonallyPeriodicBilinearUpsample(torch.nn.Module):
+    """Bilinear upsampling that enforces periodicity along the x/longitude axis."""
+
+    def __init__(self, upsampling: int | tuple[int, int] = 2):
+        super().__init__()
+        if isinstance(upsampling, tuple):
+            if tuple(upsampling) != (2, 2):
+                raise ValueError(
+                    "ZonallyPeriodicBilinearUpsample only supports 2x upsampling"
+                )
+            self.scale_h, self.scale_w = upsampling
+        else:
+            if upsampling != 2:
+                raise ValueError(
+                    "ZonallyPeriodicBilinearUpsample only supports 2x upsampling"
+                )
+            self.scale_h = self.scale_w = upsampling
```

Comment:
🐑  optional alternative
```suggestion
        if isinstance(upsampling, int):
            upsampling = (upsampling, upsampling)
        if tuple(upsampling) != (2, 2):
            raise ValueError(
                    "ZonallyPeriodicBilinearUpsample only supports 2x upsampling"
            )
        self.scale_h, self.scale_w = upsampling
```